### ISSUE TITLE
Flavors: Add `magi` flavor support for MAGI (Markdown for AI Agents) `.mda` files

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#403]
## Problem

MAGI (Markdown for Agent Guidance & Instruction) is a Markdown superset typically stored as `.mda`. It adds optional:
- YAML front matter metadata (`--- ... ---`)
- `ai-script` fenced code blocks containing a single JSON object
- Footnote relationship definitions whose payload is JSON enclosed in backticks

Today, linting `.mda` with `standard` produces avoidable noise (most commonly `MD041` when a file begins with front matter, and optionally `MD040` if users validate/allowlist fence languages and `ai-script` is treated as unknown).

## Request

Add a new flavor: `magi`, based on `standard`, with flavor-specific handling for MAGI constructs.

```toml
[global]
flavor = "magi"

[per-file-flavor]
"**/*.mda" = "magi"
```

### Auto-detection

When no flavor is configured, auto-detect:
- `.mda` -> `magi`

### YAML front matter (optional)

```yaml
---
doc-id: "38f5a922-81b2-4f1a-8d8c-3a5be4ea7511"
title: "Example"
---
```

### `ai-script` code block (JSON only)

```text
```ai-script
{
  "script-id": "summary-request-001",
  "prompt": "Summarize the previous section."
}
```
```

### Footnote relationships (JSON enclosed in backticks)

```text
See related doc.[^ref1]

[^ref1]: `{"rel-type":"parent","doc-id":"UUID-of-target","rel-desc":"Derived from primary source"}`
```

## Expected rule behavior

- `MD041`: ignore leading YAML front matter when determining the first heading.
- `MD040` (when language validation/allowlists are enabled): treat `ai-script` as an allowed fence label in `magi` flavor.

## Acceptance criteria

- `magi` is a supported `flavor` value and documented.
- `.mda` is auto-detected as `magi` when no flavor is set.
- Add fixtures/tests demonstrating:
  - `MD041` does not fire on valid `.mda` files with front matter.
  - `ai-script` fences are accepted in `magi` flavor.

## References

- https://github.com/rvben/rumdl/blob/main/docs/flavors.md
- https://docs.magi-mda.org/mdx/specification
- https://github.com/sno-ai/magi-markdown/blob/main/docs/specification.md

**Comment by rvben** [comment]
I see https://github.com/sno-ai/magi-markdown has 5 stars, so I am not sure there's enough traction yet to start supporting it.
